### DISCUSSION TITLE
Merging brush strokes without overlapping

### DISCUSSION THREAD

**Discussion by Amrita537** [fabricjs/fabric.js#9739] [Q&A]
How to merge multiple brush strokes(with opacity) without overlapping? I have tried global composition property. Couldn't achieve it. 
Image number 1: What I am getting
Image number 2: What I want 
![Screenshot 2024-03-16 170231](https://github.com/fabricjs/fabric.js/assets/47480631/698295b5-03b7-457b-8104-9eb18b82788f)

**Comment by ShaMan123** [comment]
The xor opertion suits this case
However, remember that for thw operation to work as expected both source and destination pixels must be drawn on the same context. That means the during brush drawing you will need to clip the main context or render existing paths onto the top context

**Comment by andreafalzetti** [comment]
Does anyone have a working sample? cc @Amrita537

**Comment by asturur** [comment]
If you are using a fabric brush you can try to set  globalCompositeOperation to 'xor'
